src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -76,9 +98,18 @@ public override void Initialize (AnalysisContext context)
 						operationContext.ContainingSymbol.HasAttribute (RequiresUnreferencedCodeAttribute))
 						return;
 
+					if (method.ContainingType is { } containingType && operationContext.Operation is IObjectCreationOperation)
+						CheckStaticConstructors (operationContext, containingType.StaticConstructors);
+
 					if (!method.HasAttribute (RequiresUnreferencedCodeAttribute))
 						return;
 
+					if (method.OverriddenMethod != null) {
+						for (var t = method.OverriddenMethod; t is not null && SymbolEqualityComparer.Default.Equals (method.ReturnType, t.ReturnType); t = t.OverriddenMethod) {
```

Comment:
nit: Just a matter of style but wouldn't this be clearer if you used a while-loop instead? I suppose this is to support covariant returns, maybe it'd be good to have a comment.

Reply:
I guess I only reuse one pattern I saw somewhere else in the analyzer. But you are right using the while loop might be more understandable and actually would simplify the code